src/Pages/Category/Category.jsx

```diff
@@ -1,123 +1,72 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { ProductCard } from '../../Components/Cards/ProductCard/ProductCard';
 import { useParams } from "react-router-dom";
-import PropTypes from 'prop-types';
-import { Box, Grid } from '@mui/material';
+import { Box, Grid, Typography } from '@mui/material';
 import BreadCrumb from '../../Components/BreadCrumb/BreadCrumb';
+import { fetchData } from "../../Services/network";
+import LoadingButton from '@mui/lab/LoadingButton';
 
 
-const products = [
-    {
+function Category() {
+    const { targetAPI } = useParams();
+    const [products, setProducts] = useState();
 
-        id: '1',
-        name: 'Grande',
-        descripton: 'Leather Coach Bag with adjustable starps.',
-        brand: 'Blossom Pouch',
-        price: {
-            current: '39.49',
-            actual: "78.66",
-            discountPercentage: '50'
-        },
-        rating: "4",
-        totoalRatings: '43',
+    useEffect(() => {
+        const fetchDataAsync = async () => {
+            try {
+                const result = await fetchData(`/products/${targetAPI}`);
+                setProducts(result);
+            } catch (error) {
+                console.error(error);
```

Comment:
please handle loading and errors cases correctly.
you can follow this article:
https://www.mariokandut.com/how-to-handle-errors-and-data-loading-state-with-react-hooks/